File: odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/OkRequestMode.java
package ru.ok.android.sdk;

import java.util.EnumSet;

public enum OkRequestMode {
    /**
     * Request should be signed using OAUTH access token
     */
    SIGNED,
    /**
     * Request should not be signed
     */
    UNSIGNED,
    /**
     * SDK session required
     */
    SDK_SESSION,
    /**
     * Not reporting platform via REST
     */
    NO_PLATFORM_REPORTING,
    //
    ;

    /**
     * Default request modes
     */
    public static final EnumSet<OkRequestMode> DEFAULT = EnumSet.of(OkRequestMode.SIGNED);
}

File: odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/OkRequestMode.kt
package ru.ok.android.sdk

import java.util.EnumSet

enum class OkRequestMode {
    /**
     * Request should be signed using OAUTH access token
     */
    SIGNED,
    /**
     * Request should not be signed
     */
    UNSIGNED,
    /**
     * SDK session required
     */
    SDK_SESSION,
    /**
     * Not reporting platform via REST
     */
    NO_PLATFORM_REPORTING;


    companion object {
        /**
         * Default request modes
         */
        @JvmStatic
        val DEFAULT = EnumSet.of(OkRequestMode.SIGNED)
    }
}
